pgbot/commands/admin/sudo.py

```diff
@@ -382,6 +390,98 @@ async def cmd_sudo_edit(
         await self.response_msg.delete()
         return
 
+    @add_group("sudo", "swap")
+    async def cmd_sudo_swap(
+        self,
+        msg_a: discord.Message,
+        msg_b: discord.Message,
+        embeds: bool = True,
+    ):
+        """
+        ->type More admin commands
+        ->signature pg!sudo swap <message> <message>
+        ->description Swap message contents and embeds between messages through the bot
+        ->extended description
+        Swap message contents and embeds between the two given messages.
+
+        __Args__:
+            `message_a: (Message)`
+            > A discord message whose embed
+            > should be swapped with that of `message_b`.
+
+            `message_b: (Message)`
+            > Another discord message whose embed
+            > should be swapped with that of `message_a`.
+
+            `embeds: (bool) = True`
+            > If set to `True`, the first embeds will also
+            > (when present) be swapped between the given messages.
+
+        __Raises__:
+            > `BotException`: One or more given arguments are invalid.
+            > `HTTPException`: An invalid operation was blocked by Discord.
+
+        ->example command pg!sudo swap 123456789123456789 69696969969669420
+        -----
+        """
+
+        if not utils.check_channel_permissions(
+            self.author,
+            msg_a.channel,
+            permissions=("view_channel", "send_messages"),
+        ) or not utils.check_channel_permissions(
+            self.author,
+            msg_b.channel,
+            permissions=("view_channel", "send_messages"),
+        ):
+            raise BotException(
+                f"Not enough permissions",
+                "You do not have enough permissions to run this command with the specified arguments.",
+            )
+
+        bot_member_or_user = None
+        bot_id = None
+        if self.guild is not None:
+            bot_member_or_user = self.guild.get_member(common.bot.user.id)
+            if not bot_member_or_user:
+                bot_member_or_user = await self.guild.fetch_member(common.bot.user.id)
+            bot_id = bot_member_or_user.id
+        else:
+            bot_member_or_user = common.bot.user
+            bot_id = bot_member_or_user.id
```

Comment:
why is this block of code required? `common.bot.user.id` is always defined to be user ID. and AFAIK `bot_member_or_user` is not used anywhere else too

Reply:
Yeah I guess I could eliminate that extra variable

Reply:
And the entire block, just use `common.bot.user.id` when the ID is required

Reply:
But a bot user ID does not correspond to a bot member ID

Reply:
It is the same ID

Reply:
Wait nvm